src/tests/unittests/asn1.c

```diff
@@ -47,6 +47,61 @@ static void torture_large_oid(void **state)
 	assert_int_equal(rv, SC_ERROR_NOT_SUPPORTED);
 }
 
+static void torture_oid(void **state)
+{
+	/* (without the tag and length {0x06, 0x06}) */
+	/* Small OIDs */
+	u8 small[] = {0x01, 0x02, 0x03, 0x04, 0x05, 0x06};
+	/* Limit what we can fit into the first byte */
+	u8 limit[] = {0x7F};
+	/* The second octet already oveflows to the second byte */
+	u8 two_byte[] = {0x81, 0x00};
+	/* Missing second byte, even though indicated with the first bit */
+	u8 missing[] = {0x81};
+	/* Missing second byte, even though indicated with the first bit */
+	u8 ecpubkey[] = {0x2A, 0x86, 0x48, 0xCE, 0x3D, 0x02, 0x01};
+	struct sc_object_id oid;
+	int rv = 0;
+
+	rv = sc_asn1_decode_object_id(small, sizeof(small), &oid);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(oid.value[0], 0);
+	assert_int_equal(oid.value[1], 1);
+	assert_int_equal(oid.value[2], 2);
+	assert_int_equal(oid.value[3], 3);
+	assert_int_equal(oid.value[4], 4);
+	assert_int_equal(oid.value[5], 5);
+	assert_int_equal(oid.value[6], 6);
+	assert_int_equal(oid.value[7], -1);
+
+	rv = sc_asn1_decode_object_id(limit, sizeof(limit), &oid);
+	assert_int_equal(rv, SC_SUCCESS);
+	assert_int_equal(oid.value[0], 2);
+	assert_int_equal(oid.value[1], 47);
+	assert_int_equal(oid.value[2], -1);
```

Comment:
again, separate test scenarios in the same unit test

Reply:
Resolved in the current branch